### PR TITLE
[Merged by Bors] - Make version required in connector yaml

### LINKED ISSUE #2456: Connector: Enforce version field as mandatory
[Right now in the connector yaml if the `version` field isn't specified, the connector defaults to `latest`](https://github.com/infinyon/fluvio/blob/749df76b6a277bbb3a8c966b511c2b9ad7002ca8/crates/fluvio-sc/src/k8/controllers/managed_connector_deployment.rs#L267-L274) which means that if the connector is re-created (or if the cluster is upgraded), the connector will try to get the `latest` image which may have breaking changes.

If the `version` field is unspecified, the connector should pin the version of the latest image and this image version/tag should be pinned to the version it was created with.

### PR DESCRIPTION
**Pull request by simlay**
Closes #2456.

### CODE REVIEW

**Comment by ajhunyady** [comment]
Is this change backward compatible?

**Comment by simlay** [comment]
> Is this change backward compatible?

Yes.

**Review by sehz — CHANGES_REQUESTED** [review]
is there a negative test? if required version is missing, it should fail

**Review by sehz — APPROVED** [review]
LGTM

File: crates/fluvio-controlplane-metadata/src/connector/spec.rs [hunk 1]
```diff
@@ -30,6 +30,74 @@ pub struct ManagedConnectorSpec {
...
+impl From<String> for ConnectorVersionInner {
+    fn from(inner: String) -> Self {
+        Self::String(inner)
+    }
+}
+impl ToString for ConnectorVersionInner {
+    fn to_string(&self) -> String {
+        match self {
+            ConnectorVersionInner::String(inner) => inner.to_string(),
+            ConnectorVersionInner::Option(inner) => inner.clone().unwrap_or_else(|| "".to_string()),
```
Comment (morenol) [inline crates/fluvio-controlplane-metadata/src/connector/spec.rs:58]: this is for the old connectors right? shouldn't it be `latest` by default?

Not sure what is the expected response to `fluvio connector config old-connector` when it was created with the previos version.

File: crates/fluvio-sc-schema/src/objects/list.rs [hunk 2]
```diff
@@ -27,7 +27,7 @@ where
 
 impl Request for ObjectApiListRequest {
     const API_KEY: u16 = AdminPublicApiKey::List as u16;
-    const DEFAULT_API_VERSION: i16 = 8;
+    const DEFAULT_API_VERSION: i16 = 9;
```
Comment (morenol) [inline crates/fluvio-sc-schema/src/objects/list.rs:30]: Probably we want to update this also for `watch.rs` so we don't have problems when we do the Update controller

**Review by morenol — APPROVED** [review]
looks good to me!

**Comment by simlay** [comment]
bors retry

**Comment by tjtelan** [comment]
bors retry

**Comment by tjtelan** [comment]
@simlay

The reason Bors can't merge because fluvio check test is failing.
https://github.com/infinyon/fluvio/runs/7312057421?check_suite_focus=true

**Comment by simlay** [comment]
bors retry

**Pull request closed without merging**